delivery_postlogistics/models/stock_picking.py

```diff
@@ -0,0 +1,242 @@
+# Copyright 2013-2016 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import base64
+from operator import attrgetter
+
+from odoo import _, api, exceptions, fields, models
+
+from ..postlogistics.web_service import PostlogisticsWebService
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    delivery_fixed_date = fields.Date(
+        "Fixed delivery date", help="Specific delivery date (ZAW3217)"
+    )
+    delivery_place = fields.Char(
+        "Delivery Place", help="For Deposit item service (ZAW3219)"
+    )
+    delivery_phone = fields.Char(
+        "Phone", help="For notify delivery by telephone (ZAW3213)"
+    )
+    delivery_mobile = fields.Char(
+        "Mobile", help="For notify delivery by telephone (ZAW3213)"
+    )
+
+    @api.onchange("carrier_id")
+    def onchange_carrier_id(self):
+        """ Inherit this method in your module """
+        if not self.carrier_id:
+            return
+        # This can look useless as the field carrier_code and
+        # carrier_type are related field. But it's needed to fill
+        # this field for using this fields in the view. Indeed the
+        # module that depend of delivery base can hide some field
+        # depending of the type or the code
+        carrier = self.carrier_id
+        self.update({"delivery_type": carrier.delivery_type})
+
+    def _get_packages_from_picking(self):
+        """ Get all the packages from the picking """
+        self.ensure_one()
+        operation_obj = self.env["stock.move.line"]
+        packages = self.env["stock.quant.package"].browse()
+        operations = operation_obj.search(
+            [
+                "|",
+                ("package_id", "!=", False),
+                ("result_package_id", "!=", False),
+                ("picking_id", "=", self.id),
+            ]
+        )
+        for operation in operations:
+            # Take the destination package. If empty, the package is
+            # moved so take the source one.
+            packages |= operation.result_package_id or operation.package_id
```

Comment:
these can be very slow w/ lots of records. Would you mind adding a `TODO: refactor using set of ids instead of browse records and browse at the end`?

Reply:
updated